lib/fluent/plugin/filter_viaq_data_model.rb

```diff
@@ -104,12 +216,244 @@ def delempty(thing)
       thing
     end
 
+    # map of journal fields to viaq data model field
+    JOURNAL_FIELD_MAP_SYSTEMD_T = {
+      "_AUDIT_LOGINUID"    => "AUDIT_LOGINUID",
+      "_AUDIT_SESSION"     => "AUDIT_SESSION",
+      "_BOOT_ID"           => "BOOT_ID",
+      "_CAP_EFFECTIVE"     => "CAP_EFFECTIVE",
+      "_CMDLINE"           => "CMDLINE",
+      "_COMM"              => "COMM",
+      "_EXE"               => "EXE",
+      "_GID"               => "GID",
+      "_MACHINE_ID"        => "MACHINE_ID",
+      "_PID"               => "PID",
+      "_SELINUX_CONTEXT"   => "SELINUX_CONTEXT",
+      "_SYSTEMD_CGROUP"    => "SYSTEMD_CGROUP",
+      "_SYSTEMD_OWNER_UID" => "SYSTEMD_OWNER_UID",
+      "_SYSTEMD_SESSION"   => "SYSTEMD_SESSION",
+      "_SYSTEMD_SLICE"     => "SYSTEMD_SLICE",
+      "_SYSTEMD_UNIT"      => "SYSTEMD_UNIT",
+      "_SYSTEMD_USER_UNIT" => "SYSTEMD_USER_UNIT",
+      "_TRANSPORT"         => "TRANSPORT",
+      "_UID"               => "UID"
+    }
+
+    JOURNAL_FIELD_MAP_SYSTEMD_U = {
+      "CODE_FILE"         => "CODE_FILE",
+      "CODE_FUNCTION"     => "CODE_FUNCTION",
+      "CODE_LINE"         => "CODE_LINE",
+      "ERRNO"             => "ERRNO",
+      "MESSAGE_ID"        => "MESSAGE_ID",
+      "RESULT"            => "RESULT",
+      "UNIT"              => "UNIT",
+      "SYSLOG_FACILITY"   => "SYSLOG_FACILITY",
+      "SYSLOG_IDENTIFIER" => "SYSLOG_IDENTIFIER",
+      "SYSLOG_PID"        => "SYSLOG_PID"
+    }
+
+    JOURNAL_FIELD_MAP_SYSTEMD_K = {
+      "_KERNEL_DEVICE"    => "KERNEL_DEVICE",
+      "_KERNEL_SUBSYSTEM" => "KERNEL_SUBSYSTEM",
+      "_UDEV_SYSNAME"     => "UDEV_SYSNAME",
+      "_UDEV_DEVNODE"     => "UDEV_DEVNODE",
+      "_UDEV_DEVLINK"     => "UDEV_DEVLINK",
+    }
+
+    JOURNAL_TIME_FIELDS = ['_SOURCE_REALTIME_TIMESTAMP', '__REALTIME_TIMESTAMP']
+
+    def process_journal_fields(tag, time, record, fmtr_type)
```

Comment:
I don't know how easy this class is to test, but consider moving it to a module to be mixed in.  In may facilitate unit testing